### ISSUE TITLE
Allow the usage of external pci.ids files.

### ISSUE THREAD

**Issue by gibsonpil** [gibsonpil/libpci-rs#12]
Many systems come with or often have the pci.ids file in a common location.

Allowing the file to be read from said common location, or from any external location for that matter, may be logical.

**Comment by shibedrill** [comment]
Are we referring to parsing this file at run-time or compile-time?

**Comment by gibsonpil** [comment]
Runtime a-la libpci. It would require a fair amount of extra parsing code but it could be worth it for certain use cases.

**Comment by gibsonpil** [comment]
Another advantage is that it would allow developers to make tools similar to lspci without recompiling them every time a pci.ids update comes out.

**Comment by shibedrill** [comment]
Good idea. If we choose to go that route, we can give the user the option to compile in the pci.ids file using include_bytes! or provide their own file path at compile time via an environment variable, and the file gets referenced at runtime.

**Comment by shibedrill** [comment]
I should add that if we do this right, we won't actually have to introduce any breaking changes to the API. So this doesn't necessarily have to be done prior to version 1, unless you see any major flaws in the implementation that might have to change if we alter the way we parse the data.
